task_executor/src/task_executor/base_executor.py

```diff
@@ -326,7 +371,7 @@ def prepare_task(self, task):
 
         expected_nav_duration = rospy.Duration(0)
         if self.active_task.start_node_id != '':                    
-            expected_nav_duration = self.expected_navigation_duration(task)
+            expected_nav_duration = self.expected_navigation_duration_now(self.active_task.start_node_id)
             rospy.loginfo('expected_nav_duration:  %s' % expected_nav_duration.to_sec())
```

Comment:
you can get this value from the feedback of the execute_policy action server. Not sure if you need to calculate it before calling the action though